### PR TITLE
[SPARK-31860][BUILD][2.4] only push release tags on success

### PR DESCRIPTION
**Pull request by holdenk**
### What changes were proposed in this pull request?

Only push the release tag after the build has finished.

### Why are the changes needed?

If the build fails we don't need a release tag.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Running locally with a fake user

### CODE REVIEW

**Comment by holdenk** [comment]
cc @dongjoon-hyun ?

**Comment by holdenk** [comment]
cc @shaneknapp

**Comment by dongjoon-hyun** [comment]
cc @dbtsai , too

**Comment by srowen** [comment]
I don't know these scripts well, but yes we should only push on success. This looks sensible.

**Comment by dongjoon-hyun** [comment]
Sorry, but I'll close this first to prevent accidental merge on this because `WIP` is removed.
Let's proceed this on `master` first.

**Comment by holdenk** [comment]
I've got this for the 2.4 branch since I RMd the current release, I'm annoyed you closed this @dongjoon-hyun.

**Comment by dongjoon-hyun** [comment]
This is reopened according to @holdenk 's request.

**Review by dongjoon-hyun — COMMENTED** [review]
@holdenk . In any way, I believe this help you complete 2.4.6 release. So, sorry for closing this if this made any delay to you or 2.4.6 release.

BTW, could you make a PR on `master` and `branch-3.0` eventually to prevent any regression at 3.0 after 2.4.6?

**Comment by holdenk** [comment]
Yeah I'll do a PR to master & branch-3.0 once I've gotten sign off on this being what we want to do (I know you had some reservations when we chatted).

**Comment by dbtsai** [comment]
It does make sense to push the tag only on success, and let's merge it in 2.4, and then also send out PRs to both master / 3.0. Thanks for working on this!

**Comment by holdenk** [comment]
Ok cool. I'm going to double check running it locally to make sure it the release still runs ok (I did a run last last week but just to be safe since we don't have any automated testing for these scripts), and once that's done I'll merge this and open a master & 3 PR :)

**Comment by holdenk** [comment]
jenkins retest this please

**Comment by holdenk** [comment]
Tested build script locally, works ok :)

**Comment by holdenk** [comment]
(looks like I forgot to push the commit I tested it with, pushed now)

**Comment by holdenk** [comment]
Only test failure is unit test, unrelated and different than the last test failures. I'm going to merge this to branch-2.4 now. There is a separate PR to master at https://github.com/apache/spark/pull/28700

File: dev/create-release/release-tag.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ function exit_with_usage {
   local NAME=$(basename $0)
   cat << EOF
 usage: $NAME
-Tags a Spark release on a particular branch.
+Tags a Spark release on a particular branch. Must push after
```
Comment (cloud-fan) [inline dev/create-release/release-tag.sh:27]: uncompleted sentence `Must push after ...`?

**Comment by cloud-fan** [comment]
previously I delete the tags if something goes wrong. This looks like a better solution

**Pull request closed without merging**